src/wms/src/test/java/org/geoserver/wms/map/RenderedImageMapOutputFormatExtendedTest.java

```diff
@@ -200,6 +203,97 @@ public void testRenderingTransformationChannelsSelectionFromCoverageView() throw
         map.dispose();
     }
 
+    /**
+     * Test to check that the interpolation is being propagated down to the reader when a rendering
+     * transformation is in the mix.
+     */
+    @Test
+    public void testInterpolationAppliedWithRenderingTransformation() throws Exception {
+
+        final Catalog catalog = getCatalog();
+
+        // Get the RGB-IR View which is combining an RGB GeoTIFF and an IR GeoTIFF
+        final CoverageInfo ci = catalog.getCoverageByName(RGB_IR_VIEW);
+
+        final GridCoverage2DReader reader =
+                (GridCoverage2DReader) ci.getGridCoverageReader(null, null);
+        final ReferencedEnvelope bbox = new ReferencedEnvelope(reader.getOriginalEnvelope());
+
+        final GetMapRequest request = new GetMapRequest();
+        request.setBbox(bbox);
+        request.setSRS("urn:x-ogc:def:crs:EPSG:32632");
+        request.setFormat("image/png");
+        request.setInterpolations(
+                Collections.singletonList(
+                        Interpolation.getInstance(Interpolation.INTERP_BILINEAR)));
+
+        final WMSMapContent map = new WMSMapContent(request);
+        map.setMapWidth(20);
+        map.setMapHeight(20);
+        map.setTransparent(false);
+        map.getViewport().setBounds(bbox);
+
+        // Setup a style
+        final SLDParser parser = new SLDParser(CommonFactoryFinder.getStyleFactory());
+        parser.setInput(
+                RasterSymbolizerVisitorTest.class.getResource("CropTransformAndChannelSelect.sld"));
+        final StyledLayerDescriptor sld = parser.parseSLD();
+        final NamedLayer ul = (NamedLayer) sld.getStyledLayers()[0];
+        final Style style = ul.getStyles()[0];
+
+        final Layer dl = new CachedGridReaderLayer(reader, style);
+        map.addLayer(dl);
+
+        RenderedImageMap dstImageMap = this.rasterMapProducer.produceMap(map);
+        RenderedImage destImage = dstImageMap.getImage();
+
+        assertNotNull(destImage);
+
+        Interpolation interpolation = getScalingInterpolation(destImage);
+        assertNotNull(interpolation);
+        assertEquals(interpolation, Interpolation.getInstance(Interpolation.INTERP_BILINEAR));
+        dstImageMap.dispose();
+
+        map.dispose();
+    }
+
+    private Interpolation getScalingInterpolation(RenderedImage destImage) {
+        List<RenderedOp> opImage = new ArrayList<>();
```

Comment:
This is a bit bizzarre... the test works only on the first found scale, so why not just return it directly?

Reply:
Right. Fixed